packages/@okta/vuepress-theme-prose/util/coveo/endpoint.js

```diff
@@ -1,27 +1,43 @@
 import axios from "axios";
+import jwt_decode from "jwt-decode";
+import moment from "moment";
 
 const TOKEN_ENDPOINT =
   "https://wt1ugse0be.execute-api.us-west-2.amazonaws.com/prod/token?site=developer";
 const COVEO_PIPELINE = "oktaproduction9ounvcxa";
 const COVEO_ENDPOINT = "https://platform.cloud.coveo.com/rest/search";
 
-const _getToken = async () => {
-  const {
-    data: { token }
-  } = await axios.get(TOKEN_ENDPOINT);
+const _getToken = () => {
+  const token = localStorage.getItem("coveo_token");
+  if (!token) return;
 
-  localStorage.setItem("coveo_token", token);
-  return token;
+  // Invalidate token if expired
+  const decoded = jwt_decode(token);
+  const now = moment(Date.now() / 1000);
+  const exp = moment(decoded.exp);
+  if (exp.isSameOrBefore(now)) {
```

Comment:
Can this be written as `if ( now isafter exp )`? Think it would be easier to parse

Reply:
Also, should we get a new token anyways if we are close (say, 5 minutes) from expiration?